### PR TITLE
Update changelog for 8.8 release (#10701)

### PR DESCRIPTION
**Pull request by carsonip**
Manual backport of https://github.com/elastic/apm-server/pull/10701

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 include::./changelogs/head.asciidoc[]
+include::./changelogs/8.8.asciidoc[]
```
Comment (kruskall) [inline CHANGELOG.asciidoc:2]: **question**: is this missing ? Am I missing something ?

Comment (carsonip) [reply CHANGELOG.asciidoc:1]: Nice catch, deleted head.asciidoc from this asciidoc as well.

File: changelogs/head.asciidoc
Comment (marclop) [inline changelogs/head.asciidoc:1]: Shouldn't `head.asciidoc` become `8.8.asciidoc`?

Comment (carsonip) [reply changelogs/head.asciidoc:1]: :facepalm: bad rebase, forgot to git add, thanks for catching

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (carsonip) [reply CHANGELOG.asciidoc:1]: yes 8.8 was missing in PR, only exists in unstaged area locally :facepalm:

**Review by kruskall — APPROVED** [review]

**Pull request merged**